File: app/lib/features/job_search/domain/models/job.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'job.freezed.dart';
part 'job.g.dart';

@freezed
abstract class Job with _$Job {
  const factory Job({
    required String id,
    required String source,
    required String title,
    String? company,
    String? location,
    String? description,
    String? postedDate,
    String? applicationEmail,
    String? applicationUrl,
  }) = _Job;

  factory Job.fromJson(Map<String, dynamic> json) => _$JobFromJson(json);
}

@freezed
abstract class JobSearchResult with _$JobSearchResult {
  const factory JobSearchResult({
    required List<Job> jobs,
    required int page,
    required bool hasMore,
  }) = _JobSearchResult;

  factory JobSearchResult.fromJson(Map<String, dynamic> json) =>
      _$JobSearchResultFromJson(json);
}

File: backend/src/schemas/job.ts
import { z } from "zod";

export const jobSchema = z.object({
  id: z.string(),
  source: z.literal("bundesagentur"),
  title: z.string(),
  company: z.string().optional(),
  location: z.string().optional(),
  description: z.string().optional(),
  postedDate: z.string().optional(),
  applicationEmail: z.string().optional(),
  applicationUrl: z.string().optional(),
});

export type Job = z.infer<typeof jobSchema>;

export const jobSearchResultSchema = z.object({
  jobs: z.array(jobSchema),
  page: z.number(),
  hasMore: z.boolean(),
});

export type JobSearchResult = z.infer<typeof jobSearchResultSchema>;
